Inc/HALAL/Services/Encoder/Encoder.hpp

```diff
@@ -0,0 +1,120 @@
+/*
+ * Encoder.hpp
+ *
+ *  Created on: 27 oct. 2022
+ *      Author: Pablo
+ */
+
+#pragma once
+#ifdef HAL_TIM_MODULE_ENABLED
+#include "ST-LIB.hpp"
+
+/**
+ * @struct DoublePin
+ * @brief Estruct that contains a pair of pins
+ *
+ * @var DoublePin::pin1
+ * The first pin of the structure
+ * @var DoublePin::pin2
+ * The second pin of the structure
+ *
+ * */
+struct DoublePin {
+	Pin* pin1;
+	Pin* pin2;
+
+	bool operator<(const DoublePin& other) const {
+		if (pin1 < other.pin1) {
+			return true;
+		}else if(pin1 == other.pin1){
+			if (pin2 < other.pin2) {
+				return true;
+			}
+		}
+		return false;
+	}
+
+	bool operator<(const DoublePin* other) const {
+		if (*pin1 < *other->pin1) {
+			return true;
+		}else if(*pin1 == *other->pin1){
+			if (*pin2 < *other->pin2) {
+				return true;
+			}
+		}
+		return false;
+	}
+
+	bool operator==(const DoublePin& other) const {
+		return pin1 == other.pin1 && pin2 == other.pin2;
+	}
+
+	bool operator==(const DoublePin* other) const {
+			return *pin1 == *other->pin1 && *pin2 == *other->pin2;
+		}
+};
+
+struct EncoderPinTimer{
+	Pin pin1;
+	Pin pin2;
+	TIM_HandleTypeDef* timer;
+};
+
+/**
+ * @brief Encoder service class for abstracting the use of the encoder with the HAL library.
```

Comment:
Short sentences are more legible.
```suggestion
 * @brief Encoder service class. Abstracts the use of the encoder with the HAL library.
```